### ISSUE TITLE
Nav doc: when is epub:type necessary

### ISSUE THREAD

**Issue by mattgarrish** [w3c/epub-specs#976]
Section 5.4.1 currently reads:

> When a nav element carries the epub:type attribute [Content Docs 3.1] in an EPUB Navigation Document, this specification restricts the content model of the element and its descendants as follows:

http://www.idpf.org/epub/31/spec/epub-packages.html#sec-package-nav-def-model

But then in 5.4.2.5 it says:

> EPUB Navigation Documents MAY include one or more nav elements in addition to the toc, page-list and landmarks nav elements defined in the preceding sections. These additional nav elements SHOULD have an epub:type attribute to provide a machine-readable semantic, and MUST have a human-readable heading as their first child.

http://www.idpf.org/epub/31/spec/epub-packages.html#sec-package-nav-def-types-other

The problem is that if you omit epub:type to avoid the content model restrictions, you're supposed to get a warning. For most people, that will mean the content model restrictions cannot be avoided.

The navigation document is also supposed to be a content document that allows any kind of additional content within it, but these requirements prevent that from truly happening.

Since machine-processing of non-standardized nav elements seems unlikely, can we drop the "should" requirement from other navs and instead shift it so that reading systems should ignore nav elements that don't have an epub:type element or carry a semantic they don't recognize? (Somewhat overlapping here with #975.)

**Comment by rdeltour** [comment]
I agree there's a spec issue here. If only for the heterogeneous content restriction rules: on the one hand, content model restrictions only apply to `nav` elements with `epub:type`; on the other hand we still have content model restrictions –but only about first-child headings– for `nav` elements with no `epub:type` attribute.

I'd rather see content model restrictions applying either uniformly to all `nav` in the Nav Doc with an `epub:type` attribute, and totally ignored on any other `nav`.

In other words, I suggest to change 5.4.2.5 to:

> These additional nav elements MUST have an epub:type attribute

and add prose to explain that non-semantically-enriched `nav` may be ignored by reading systems (for navigational purposes; they'd still have to be rendered when the doc is processed like any other Content Doc).

**Comment by mattgarrish** [comment]
That works for me. I'd even drop the requirement for a human-readable heading to be consistent with the content model defined above.

Perhaps:

> EPUB Navigation Documents MAY include one or more nav elements in addition to the toc, page-list and landmarks nav elements defined in the preceding sections. If these nav elements are intended for Reading System processing, they MUST have an epub:type attribute with a machine-readable semantic and are subject to the content model restrictions defined in 5.4.1 The nav Element: Restrictions.
> 
> This specification imposes ...
> 
> Reading System support for all additional nav elements is OPTIONAL.

(Depending on the resolution to #975, this latter statement could be moved up into the reading system requirements in parallel to required support.)

**Comment by GeorgeKerscher** [comment]
It seems that a person should know where a link is taking them. The nav doc could become so large that it would not be possible to traverse. So, I agree with Matt.
